apps/service_providers/llm_service/adapters.py

```diff
@@ -194,158 +323,11 @@ def get_messages_to_sync_to_thread(self):
 
     def get_metadata(self, key: Chat.MetadataKeys):
         """Chat metadata"""
-        return self.chat.get_metadata(key)
+        return self.session.chat.get_metadata(key)
```

Comment:
I think this is only used for getting the thread ID - we could rename it to `get_thread_id` or make it a property `thread_id`:

```python
    @property
    def thread_id(self):
        return self.session.chat.get_metadata(Chat.MetadataKeys.OPENAI_THREAD_ID)

    @thread_id.setter
    def set_thread_id(self, value):
        key = Chat.MetadataKeys.OPENAI_THREAD_ID
        if self.trace_service:
            self.trace_service.update_trace({key: value})
        self.session.chat.set_metadata(key, value)
```

Reply:
Good idea. I'll update

Reply:
I think this should work:

```python
setter_mock = Mock()
mock_property = ChatAdapter.thread_id.setter(setter_mock)
with patch.object(ChatAdapter, 'thread_id', mock_property):
    adapter.thread_id = 1
    setter_mock.assert_called_with(adapter, 1)
```

Reply:
Hmm, that didn't work either. I might have done it wrong, I don't know. I went with [this approach](https://github.com/dimagi/open-chat-studio/pull/918/commits/64c8818aea6cba386689298005c14d48929be11f) rather to save time, which is suboptimal, since I'm writing code to accomodate a test, but at least we're testing that the `thread_id` was called again. Feel free to take a swing at it if you'd like.